# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gcc/gcc2/protector-docs.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


Add documentation (man and info) to protector.patch.

--- ./gcc/gcc.1.orig	Tue Oct 16 08:19:36 2001
+++ ./gcc/gcc.1	Tue Oct 16 08:27:46 2001
@@ -515,6 +515,7 @@
 \-fshared\-data
 \-fshort\-enums
 \-fshort\-double
+\-fstack\-protector
 \-fvolatile
 \-fvolatile\-global
 \-fverbose\-asm
@@ -3856,6 +3857,12 @@
 .B no\-\c
 \&\|' or adding
 it.
+.TP
+.B \-fstack\-protector
+Enable the Stack-Smashing Protector (
+.B http://www.trl.ibm.com/projects/security/ssp/
+). This option is only available in ROCK Linux (and probably
+some other distributions). It's not a standard GCC option.
 .TP
 .B \-fnonnull\-objects
 Assume that objects reached through references are not null
--- ./gcc/invoke.texi.orig	Tue Oct 16 08:38:25 2001
+++ ./gcc/invoke.texi	Tue Oct 16 08:38:17 2001
@@ -421,8 +421,8 @@
 -fno-common  -fno-ident  -fno-gnu-linker
 -fpcc-struct-return  -fpic  -fPIC
 -freg-struct-return  -fshared-data  -fshort-enums
--fshort-double  -fvolatile  -fvolatile-global -fvolatile-static
--fverbose-asm -fpack-struct  -fstack-check
+-fshort-double  -fstack-protector  -fvolatile  -fvolatile-global
+-fvolatile-static  -fverbose-asm -fpack-struct  -fstack-check
 -fargument-alias  -fargument-noalias
 -fargument-noalias-global
 -fleading-underscore
@@ -6147,6 +6147,12 @@
 it.
 
 @table @code
+@item -fstack-protector
+Enable the Stack-Smashing Protector
+(http://www.trl.ibm.com/projects/security/ssp/).
+This option is only available in ROCK Linux (and probably
+some other distributions). It's not a standard GCC option.
+
 @item -fexceptions
 Enable exception handling. Generates extra code needed to propagate
 exceptions.  For some targets, this implies generation of frame unwind 
